src/magic-square.ts

```diff
@@ -0,0 +1,100 @@
+import * as ethersAddressUtils from '@ethersproject/address';
+import * as ethersHashUtils from '@ethersproject/hash';
+import * as ethersStringsUtils from '@ethersproject/strings';
+import * as ethersTxUtils from '@ethersproject/transactions';
+import { validateAddress, ValidationResult, verifySignature, getPkhfromPk } from '@taquito/utils';
+import { StatusCodes } from 'http-status-codes';
+
+import { objectStorageMethodsFactory } from './redis';
+import { CodedError } from './utils/errors';
+import { safeCheck } from './utils/helpers';
+import { getSigningNonce, removeSigningNonce } from './utils/signing-nonce';
+
+interface Participant {
+  pkh: string;
+  evmPkh: string;
+  ts: string;
+}
+
+const REDIS_DB_KEY = 'magic_square_quest';
+
+const redisStorage = objectStorageMethodsFactory<Participant, null>(REDIS_DB_KEY, null);
+
+export function getMagicSquareQuestParticipants() {
+  return redisStorage.getAllValues().then(records => Object.values(records));
+}
+
+interface StartQuestPayload {
+  pkh: string;
+  publicKey: string;
+  messageBytes: string;
+  signature: string;
+  evm: {
+    pkh: string;
+    messageBytes: string;
+    signature: string;
+  };
+}
+
+export async function startMagicSquareQuest({ pkh, publicKey, messageBytes, signature, evm }: StartQuestPayload) {
+  // Public Key Hashes
+
+  if (!safeCheck(() => validateAddress(pkh) === ValidationResult.VALID && getPkhfromPk(publicKey) === pkh))
```

Comment:
What is the point behind passing both `pkh` and `publicKey` as arguments to the `startMagicSquareQuest(...)` function when it's possible to obtain `pkh` directly from `publicKey`? Looks like `publicKey` only might be sufficient